transactions/transactor.go

```diff
@@ -69,6 +69,97 @@ func (t *Transactor) SendTransaction(sendArgs SendTxArgs, verifiedAccount *accou
 	return
 }
 
+// SendTransactionWithSignature receive a transaction and a signature, serialize them together and propage it to the network.
+// It's different from eth_sendRawTransaction because it receives a signature and not a serialized transaction with signature.
+// Since the transactions is already signed, we assume it was validated and used the right nonce.
+func (t *Transactor) SendTransactionWithSignature(args SendTxArgs, sig []byte) (hash gethcommon.Hash, err error) {
```

Comment:
I have an impression that this method is already defined somewhere or one very similar to it.

Reply:
@adambabik I think we have SendRawTransaction/SendSignedTransaction right? but it needs the full signed transaction and not only the signature. but from the keycard we receive only a signature and a tx not serialized

Reply:
Right, I am just thinking if we can generalize it or put some fragments into a separate function/method. Especially, the code around getting params and deciding whether we create a tx or contract. Also, getting a final nonce is tricky and it would be nice to encapsulate this logic.

Reply:
@adambabik it's very similar to SendTransaction but:
- it doesn't set the nonce in case the remote nonce provider returns a higher one
- it doesn't sign the transaction because we already have the signature
- it doesn't alter any tx args because the signature would be invalid

Reply:
@adambabik ok I can refactor the tx creation, but the nonce part is different so I would keep that separated for the 2 cases, what do you think?

Reply:
I see, `SendTransaction` is highly specified with a ton of tx args magic stuff. Ok, I guess there is no really a way to do something better here without refactoring `SendTransaction` which we don't want to do.